deployments/local-minio/deploy.sh

```diff
@@ -1,64 +1,118 @@
 #!/usr/bin/env bash
 
 # Name to give the docker network and stack. Update network instances in docker-compose.yml file if modified.
-networkname="cdr_local-minio"
+name="cdr_local-minio"
 
+helptext="\
+Usage: deploy.sh [command]
+Utility script for setting up a docker network and deploying a local instance of CDR into a docker swarm.
+'deploy' is the default option if no command is given.
+
+Commands:
+  d, deploy\t\tcreate the docker network and stack, stopping if one already exists
+  u, update\t\tattempts an in-place update of the deployed docker stack
+  c, clean\t\tremoves the deployed docker stack and network (destructive)
+  r, redeploy\t\tclean + deploy: completely removes the stack and network and deploys new instances (destructive)
+"
+
+# Removes the docker stack and network.
+# Docker networks usually take a few seconds to go down, so it waits to prevent redeployment issues
+function cleanup () {
+    printf "Removing $name docker stack and network...\n"
+    docker stack rm $name
+    docker network rm $name
+    waiting=$(docker network ls | grep "$name")
+    printf "\nWaiting for docker network '$name' to go down."
```

Comment:
:pencil2: There are two while loops. One to wait for network termination and one to wait for containers to come up. Define a waiting function: 

```
# Evaluate the first parameter passed to the function every periodically.
# When the expression evaluates to an empty string, return.
function wait_for_empty_results() {
  sleep 1
  while [[ -n  $(eval $1) ]]; do
    sleep 1
    printf "." >&2
  done
}
```

Then waiting for the network to be destroyed is:
    `wait_for_empty_results "docker network ls | grep $name"`

And waiting for the containers
    `wait_for_empty_results "docker service ls | grep $name_.*0/[[:digit:]]+"`